config/v1/types_infrastructure.go

```diff
@@ -22,6 +22,11 @@ type Infrastructure struct {
 type InfrastructureSpec struct {
 	// secret reference?
 	// configmap reference to file?
+
+	// platform contains settings specific to the infrastructure providers supported
+	// by the given infrastructure.
+	// +optional
+	Platform PlatformSpec `json:"platform,omitempty"`
```

Comment:
Does it make sense to have this as a pointer, in order to make it truly optional?

Reply:
> Does it make sense to have this as a pointer, in order to make it truly optional?

I'd prefer a pointer.